lib/dbservice_web/controllers/session_controller.ex

```diff
@@ -27,8 +29,23 @@ defmodule DbserviceWeb.SessionController do
     response(200, "OK", Schema.ref(:Sessions))
   end
 
-  def index(conn, _params) do
-    session = Sessions.list_session()
+  def index(conn, params) do
+    session =
+      Enum.reduce(params, Session, fn
+        {"platform_link", platform_link}, query ->
+          from(u in query, where: u.platform_link == ^platform_link)
+
+        {"portal_link", portal_link}, query ->
+          from(u in query, where: u.portal_link == ^portal_link)
+
+        {"session_id", session_id}, query ->
+          from(u in query, where: u.session_id == ^session_id)
+
+        _, query ->
+          query
+      end)
```

Comment:
Can't we do something like this? I've written pseudo code but hopefully it's straightforward and easy to understand
```
if (params) {
   foreach (param in params) {
     query = {param, param_value}, query -> from(u in query, where u[param] = param_value)
   }
}
```

So the idea is not hardcode any columns and instead just pass whatever key & value we're sending in the API route